default.nix

```diff
@@ -787,6 +787,83 @@ rec {
       '';
   };
 
+  # Helper function to filter tests by type
+  filter_tests = type: tests:
+    let
+      # Get all test names that match the pattern
+      debug_tests = builtins.filter (name: 
+        builtins.match ".*-dbg$" name != null || 
+        builtins.match ".*-debug$" name != null
+      ) (builtins.attrNames tests);
+      
+      # Get all test names that don't match the pattern
+      release_tests = builtins.filter (name:
+        builtins.match ".*-dbg$" name == null && 
+        builtins.match ".*-debug$" name == null
+      ) (builtins.attrNames tests);
+      
+      # Select which set of names to use
+      selected_names = if type == "debug" then debug_tests else release_tests;
+    in
+      # Get the actual derivations for the selected names
+      builtins.map (name: tests.${name}) selected_names;
+
+  # Release version - excludes debug tests
+  release-systems-go = nixpkgs.releaseTools.aggregate {
+    name = "release-systems-go";
+    constituents = [
```

Comment:
please use a `let` to use a common list (below too).

Reply:
like that? I am not sure I properly understood the ask.

Reply:
and I am also not sure that what I did now actually works

Reply:
Just factoring it out. I have a commit here, can I push?

Reply:
thanks a lot!